exporters/metric/cortex/config.go

```diff
@@ -0,0 +1,104 @@
+package cortex
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/spf13/viper"
+)
+
+// Config contains properties the Exporter uses to export metrics data to Cortex.
+type Config struct {
+	Endpoint        string            `mapstructure:"url"`
+	RemoteTimeout   string            `mapstructure:"remote_timeout"`
+	Name            string            `mapstructure:"name"`
+	BasicAuth       map[string]string `mapstructure:"basic_auth"`
+	BearerToken     string            `mapstructure:"bearer_token"`
+	BearerTokenFile string            `mapstructure:"bearer_token_file"`
+	TLSConfig       map[string]string `mapstructure:"tls_config"`
+	ProxyURL        string            `mapstructure:"proxy_url"`
+	PushInterval    string            `mapstructure:"push_interval"`
+	Client          *http.Client
+}
+
+// Option sets an option for a Config struct.
+type Option interface {
+	Apply(*Config)
+}
+
+// WithFilepath adds a path where Viper will search for the YAML file in.
+func WithFilepath(filepath string) Option {
+	return filepathOption(filepath)
+}
+
+type filepathOption string
+
+func (o filepathOption) Apply(config *Config) {
+	viper.AddConfigPath(string(o))
+}
+
+// WithClient adds a custom http.Client to the Config struct.
+func WithClient(client *http.Client) Option {
+	return clientOption{client}
+}
+
+type clientOption struct {
+	client *http.Client
+}
+
+func (o clientOption) Apply(config *Config) {
+	config.Client = (*http.Client)(o.client)
+}
+
+// NewConfig creates a Config struct with a YAML file and applies Option functions to the Config
+// struct.
+func NewConfig(filename string, opts ...Option) (*Config, error) {
```

Comment:
Are there further sane defaults we can provide through this function for configuration options such as `Endpoint`, `Client`, etc.?